### PR TITLE
Feature/dog/digs

### PR DESCRIPTION
**Pull request by ptPierre**

### CODE REVIEW

File: schema.graphql [hunk 1]
```diff
@@ -397,6 +399,13 @@ type SystemState @entity {
   " Total supply of ERC20.DAI "
   daiTotalSupply: BigDecimal!
 
+  #
+  #Liquidation 2.0 parameters (Dog)
+  #
+
+  " Amount of DAI needed to cover debt + fees of active auctions (Dog#Dirt) "
+  totalAuctionDebtAndFees: BigDecimal!
```
Comment (llopez) [inline schema.graphql]: maybe `totalDaiAmountToCoverDebtAndFees`? what do you think?

File: schema.graphql [hunk 2]
```diff
@@ -91,7 +91,9 @@ type CollateralType @entity {
   # TODO : clip;  // Liquidator
   # TODO : clip;  // Liquidator
   # TODO: uint256 hole;  // Max DAI needed to cover debt+fees of active auctions per ilk [rad]
-  # TODO: uint256 dirt;  // Amt DAI needed to cover debt+fees of active auctions per ilk [rad]
+
+  " Amt DAI needed to cover debt+fees of active auctions per ilk [rad] "
+  dirt: BigDecimal!
```
Comment (llopez) [inline schema.graphql]: improve naming, maybe: `daiAmountToCoverDebtAndFees`

File: src/mappings/modules/liquidation/dog.ts [hunk 3]
```diff
@@ -10,3 +13,14 @@ export function handleCage(event: Cage): void {
...
   log.save()
 }
+
+export function handleDigs(event: Digs): void {
+  let systemState = system.getSystemState(event)
+  let amount = units.fromRad(event.params.rad)
+  systemState.totalAuctionDebtAndFees = systemState.totalAuctionDebtAndFees.minus(amount)
+  systemState.save()
+
+  let collType = collateralTypes.loadOrCreateCollateralType(event.params.ilk.toString())
```
Comment (llopez) [inline src/mappings/modules/liquidation/dog.ts]: variable name to `collateralType`

**Review by llopez — CHANGES_REQUESTED** [review]

File: tests/mappings/modules/liquidation/dog/handleDigs.test.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    let collateralType = new CollateralType("ETH-A")
+    collateralType.dirt = BigDecimal.fromString('0.0')
+    collateralType.save()
+
+    mockDebt()
+
+    handleDigs(event)
+
+    assert.fieldEquals('SystemState', 'current', 'totalAuctionDebtAndFees', '-10000')
+    assert.fieldEquals('CollateralType', 'ETH-A', 'dirt', '-10000')
```
Comment (rtomas) [inline tests/mappings/modules/liquidation/dog/handleDigs.test.ts]: Do you think is a good idea to add to the entity systemState in the field totalAuctionDebtAndFees = 0 and save it ?

File: tests/mappings/modules/liquidation/dog/handleDigs.test.ts [hunk 5]
```diff
@@ -0,0 +1,37 @@
+import { Bytes, BigInt, BigDecimal } from '@graphprotocol/graph-ts'
+import { describe, test, assert, clearStore, beforeAll, afterAll } from 'matchstick-as'
+import { Digs } from '../../../../../generated/Dog/Dog'
+import { handleDigs } from '../../../../../src/mappings/modules/liquidation/dog'
+import { tests } from '../../../../../src/mappings/modules/tests'
+import { mockDebt } from '../../../../helpers/mockedFunctions'
+import { CollateralType } from '../../../../../generated/schema'
+
+function createEvent(): Digs {
+    let ilk = Bytes.fromHexString("4554482D41000000000000000000000000000000000000000000000000000000")
```
Comment (llopez) [inline tests/mappings/modules/liquidation/dog/handleDigs.test.ts]: this is a little hard to read for humans. maybe plain string value?

**Review by llopez — APPROVED** [review]

**Pull request merged**